src/controllers/mainController.ts

```diff
@@ -1782,6 +1787,191 @@ export default class MainController implements vscode.Disposable {
             ),
         );
 
+        // Data-tier Application - Main command
+        this._context.subscriptions.push(
```

Comment:
Looks like there's a bunch of duplicated code between these registrations; can you shift it into a helper function?